swift/extractor/SwiftDispatcher.h

```diff
@@ -120,6 +132,33 @@ class SwiftDispatcher {
                    locatableLabel);
   }
 
+  // In order to not emit duplicated entries for declarations, we restrict emission to only
+  // Decls declared within the current "scope".
+  // Depending on the SwiftExtractionMode the scope is defined as follows:
+  //  - SwiftExtractionMode::Module: the current scope means the current module. This is used in
+  //    the case of system or builtin modules.
+  //  - SwiftExtractionMode::PrimaryFile: in this mode, we extract several files belnoging to the
+  //    same module one by one. In this mode, the restrict emission only to the same file ignoring
+  //    all the other files.
+  bool shouldEmitDeclBody(swift::Decl* decl) {
+    switch (extractionMode) {
+      case SwiftExtractionMode::Module: {
+        return currentModule == decl->getModuleContext();
+      } break;
+      case SwiftExtractionMode::PrimaryFile: {
+        swift::SourceLoc location = decl->getStartLoc();
+        if (!location.isValid()) {
+          return false;
+        }
```

Comment:
now that I think about it, this might exclude some compiler synthesized declarations, which we currently explicitly exclude from tests (which is probably not good).

This is probably not something we can really do...

We might need maybe to climb the `ASTContext` tree until we have a valid location and then use that.

and we need to have a test showing synthesized stuff, maybe tweaking the test inclusion predicate

Reply:
That's a good point! What about checking for filename only if the location is present, otherwise emitting it under condition that the module is the same? 🤔

Reply:
(by another file I mean another file of the same module we are currently building)

Reply:
Yes, you are right!
I'll add a TODO here, and then when we have an example we can extend the functionality. WDYT?

Reply:
ah, whoops, now the enum test actually shows this problem :hand_over_mouth: If you want to solve it at a later stage, you can disable the enum test by renaming `enums.swift` to `enum.swift.TODO` and running codegen again.